GitUI/CommandsDialogs/FormCreateTag.cs

```diff
@@ -66,21 +69,13 @@ private string CreateTag()
                 MessageBox.Show(this, _noRevisionSelected.Text, _messageCaption.Text);
                 return string.Empty;
             }
-            if (annotate.Checked)
-            {
-                if (string.IsNullOrEmpty(tagMessage.Text))
-                {
-                    MessageBox.Show(this, _noTagMessage.Text, _messageCaption.Text);
-                    return string.Empty;
-                }
-
-                File.WriteAllText(Path.Combine(Module.GetGitDirectory(), "TAGMESSAGE"), tagMessage.Text);
-            }
-
-            var s = Module.Tag(textBoxTagName.Text, revision, annotate.Checked, ForceTag.Checked);
 
+            GitTagController.TagOperation _tagOperation = (GitTagController.TagOperation) annotate.SelectedIndex;
```

Comment:
this doesn't look type-safe.
perhaps a better way is to create a method `private TagOperation GetSelectedOperation()` which extracts the selected operation from the dropdown in a type-safe manner.
This will also make `AnnotateDropDownChanged` more readable and maintainable.